src/Stratis.Bitcoin.Features.MemoryPool/MempoolValidator.cs

```diff
@@ -1043,7 +1043,7 @@ private void CheckAllInputs(MempoolValidationContext context)
             Transaction tx = context.Transaction;
             if (!context.Transaction.IsCoinBase)
             {
-                this.consensusValidator.CheckInputs(context.Transaction, context.View.Set, this.chain.Height + 1);
+                this.consensusRules.GetRule<PowCoinviewRule>().CheckInputs(context.Transaction, context.View.Set, this.chain.Height + 1);
```

Comment:
Logic here is changed. Before when we were getting PowCoinviewRule and we are running PoS node we would get an instance which has overriden CheckInputs method. But now we are getting it always from PowCoinviewRule so even in case we are running PoS node we will still be checking according to the PoW rules for CheckInputs

Reply:
I think the confusion is that it is called `PowCoinviewRule` but since when running a PoS node, only the `PosCoinviewRule` is registered, and that inherits from PowCoinviewRule - I believe it would actually get the PosCoinviewRule as expected and use the `CheckInputs` from there.    Did you actually experience this behaviour when debugging it or just from eyeballing the code?

Reply:
Andy did also suggest that and in order to have minimal changes for this PR I preferred to not use a base.
If we do a base there will be other parts in the coinview rule we can fix.
So this is fine for now and besides this will be refactored when we do the mempool rules.

Reply:
I think that because of the minimal changes we will have plenty of bad code inserted in the code base and it will not get fixed later because we just forget about it. It is not the case of this particular code which you are correct will be Refactored for sure, but in other cases and afraid no one holds the to do list, so I would expect many things to be just forgotten.

Reply:
I absolutely agree, we need to take that list and promote it to the board, and tag it is sp2 so that they are candidates for the next sprint

Reply:
It looks like some of the comments were already overwritten. We would probably need to go back in emails or I don't know how to recover the list. There should be an issue created with a list of follow-ups immediately when you decide to skip some comment, otherwise it gets forgotten.

Reply:
what do you mean by skip some comment? the PR comment? or code comment?

Reply:
was either commit comment or something, but i can rememember at least a couple of comments that were not implemented nor said not to be implemented

this is one of the things
https://github.com/stratisproject/StratisBitcoinFullNode/pull/1414#discussion_r189283428

others i can't even find anymore :-( we spent quite a lot of time on a review making a lot of comments and it looks like they disappeared or something

Reply:
oh, here is another one https://github.com/stratisproject/StratisBitcoinFullNode/pull/1414#discussion_r189021413

Reply:
ok, found them now, basically just ignored, they did not disappear, just hard to find - https://github.com/stratisproject/StratisBitcoinFullNode/pull/1414#discussion_r189019667

https://github.com/stratisproject/StratisBitcoinFullNode/pull/1414#discussion_r189020082

maybe there are more, not sure

Reply:
Yes @majikandy please make sure you have responded to every comment either that you agree (and made the change) or start a discussion so nothing is forgotten.